esphome/components/bedjet/bedjet_child.h

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include "bedjet_codec.h"
+
+namespace esphome {
+namespace bedjet {
+
+// Forward declare BedJetHub
+class BedJetHub;
+
+class BedJetClient {
+ public:
+  void register_parent(BedJetHub *parent);
+
+  virtual void on_status(const BedjetStatusPacket *data) = 0;
+  virtual void on_bedjet_state(bool is_ready) = 0;
+
+ protected:
+  friend BedJetHub;
+  virtual std::string describe() = 0;
+  BedJetHub *parent_{};
+};
```

Comment:
This will allow you to remove the `parent` code inline in this class.
Also `#include "esphome/core/helpers.h"`


```suggestion
class BedJetClient : public Parented<BedJetHub> {
 public:
  virtual void on_status(const BedjetStatusPacket *data) = 0;
  virtual void on_bedjet_state(bool is_ready) = 0;

 protected:
  friend BedJetHub;
  virtual std::string describe() = 0;
};
```

Reply:
Ahh I did not know about `Parented`, thanks! I'll make that change shortly.

Reply:
Done